extensions/guacamole-auth-sso/modules/guacamole-auth-sso-saml/src/main/java/org/apache/guacamole/auth/saml/conf/ConfigurationService.java

```diff
@@ -371,7 +371,13 @@ public int getAuthenticationTimeout() throws GuacamoleException {
      *     If the X.509 certificate cannot be parsed.
      */
     public File getCertificateFile() throws GuacamoleException {
-        return environment.getProperty(SAML_X509_CERT_PATH);
+    	File certificate = null;
+    	try {
+			certificate = environment.getProperty(SAML_X509_CERT_PATH).getCanonicalFile();
+		} catch (IOException | GuacamoleException e) {			
+			e.printStackTrace();
```

Comment:
If an error should be logged, it should be logged by a `Logger` instance with the stack trace logged only at the debug level. We typically do this by logging things twice, with an admin-facing message at the error level and a developer-facing message at the debug level.

That said, I think the proper thing to do here would be to just rethrow the `IOException` as a `GuacamoleException`, with the `GuacamoleException` adding a meaningful message.